plugins/inputs/nftables/nftable.go

```diff
@@ -0,0 +1,100 @@
+//go:build linux
+
+package nftables
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type nftable struct {
+	Metainfo          *Metainfo
+	Rules             []*Rule
+	JSONSchemaVersion int `json:"json_schema_version"`
+}
+
+// UnmarshalJSON handles custom parsing of the nftables JSON output which uses
+// a generic structure incompatible that standard JSON unmarshaling.
+func (nftable *nftable) UnmarshalJSON(b []byte) error {
+	var rawTable map[string][]map[string]json.RawMessage
+	if err := json.Unmarshal(b, &rawTable); err != nil {
+		return fmt.Errorf("unable to unmarshal: %s", b)
+	}
+	// []map[string]interface
+	nfthings := rawTable["nftables"]
+	for _, nfthing := range nfthings {
+		hasKey := func(key string) bool { _, ok := nfthing[key]; return ok }
+		switch {
+		case hasKey("metainfo"):
+			var mi Metainfo
+			err := json.Unmarshal(nfthing["metainfo"], &mi)
+			if err != nil {
+				return fmt.Errorf("unable to parse metadata: %w", err)
+			}
+			nftable.Metainfo = &mi
+		case hasKey("rule"):
+			var r Rule
+			err := json.Unmarshal(nfthing["rule"], &r)
+			if err != nil {
+				return fmt.Errorf("unable to parse rule: %w", err)
+			}
+			nftable.Rules = append(nftable.Rules, &r)
+		default:
+			// something we are not parsing
+		}
+	}
+	return nil
+}
+
+type Metainfo struct {
```

Comment:
You do not need to export those data-structures! Same for the other types below.